lib/elixir/lib/access.ex

```diff
@@ -700,4 +700,158 @@ defmodule Access do
   defp get_and_update_at([], _index, _next, updates) do
     {nil, :lists.reverse(updates)}
   end
+
+  @doc ~S"""
+    Returns a function that accesses all elements of a list that match the provided predicate.
+
+    The returned function is typically passed as an accessor to `Kernel.get_in/2`,
+    `Kernel.get_and_update_in/3`, and friends.
+
+    ## Examples
+
+        iex> list = [%{name: "john", salary: 10}, %{name: "mary", salary: 20}, %{name: "francine", salary: 30}]
```

Comment:
We can probably do it with a list of two to simplify examples.

Reply:
I ended up going with a list of 3 items to show order preservation, and specifically with `filter` to show how order is preserved on both sides. I don't mind changing it though, if you feel strongly about it.

Reply:
I could also write a non-example test to confirm that, and use the simpler example?

Reply:
> I could also write a non-example test to confirm that, and use the simpler example?

Perfect.